Tests/SourceKitTests/ImplementationTests.swift

```diff
@@ -0,0 +1,67 @@
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@testable import SourceKit
+import LanguageServerProtocol
+import XCTest
+import SKTestSupport
+import ISDBTestSupport
+
+final class ImplementationTests: XCTestCase {
+  func testImplementation() throws {
+    let ws = try staticSourceKitTibsWorkspace(name: "Implementation")!
+    try ws.buildAndIndex()
+
+    try ws.openDocument(ws.testLoc("a.swift").url, language: .swift)
+    try ws.openDocument(ws.testLoc("b.swift").url, language: .swift)
+
+    func impls(at testLoc: TestLocation) throws -> [Location] {
+      let textDocument = TextDocumentIdentifier(testLoc.url)
+      let request = ImplementationRequest(textDocument: textDocument, position: Position(testLoc))
+      let implementations = try ws.sk.sendSync(request)
+      return implementations
+    }
+    func testLoc(_ name: String) -> TestLocation {
+      ws.testLoc(name)
+    }  
+    func loc(_ name: String) -> Location {
+      Location(ws.testLoc(name))
+    }
+    
+    try XCTAssertEqual(impls(at: testLoc("Protocol")), [loc("StructConformance")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolStaticVar")), [loc("StructStaticVar")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolStaticFunction")), [loc("StructStaticFunction")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolVariable")), [loc("StructVariable")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolFunction")), [loc("StructFunction")])
+    try XCTAssertEqual(impls(at: testLoc("Class")), [loc("SubclassConformance")])
+    try XCTAssertEqual(impls(at: testLoc("ClassClassVar")), [loc("SubclassClassVar")])
+    try XCTAssertEqual(impls(at: testLoc("ClassClassFunction")), [loc("SubclassClassFunction")])
+    try XCTAssertEqual(impls(at: testLoc("ClassVariable")), [loc("SubclassVariable")])
+    try XCTAssertEqual(impls(at: testLoc("ClassFunction")), [loc("SubclassFunction")])
+
+    try XCTAssertEqual(impls(at: testLoc("Sepulcidae")), [loc("ParapamphiliinaeConformance"), loc("XyelulinaeConformance"), loc("TrematothoracinaeConformance")])
+    try XCTAssertEqual(impls(at: testLoc("Parapamphiliinae")), [loc("MicramphiliusConformance"), loc("PamparaphiliusConformance")])
+    try XCTAssertEqual(impls(at: testLoc("Xyelulinae")), [loc("XyelulaConformance")])
+    try XCTAssertEqual(impls(at: testLoc("Trematothoracinae")), [])
+
+    try XCTAssertEqual(impls(at: testLoc("Prozaiczne")), [loc("MurkwiaConformance2"), loc("SepulkaConformance1")])
+    // FIXME: non-ascii characters break Tibs `TestLocation`s
+    // try XCTAssertEqual(impls(at: testLoc("Sepulkowate")), [loc("MurkwiaConformance1"), loc("SepulkaConformance2"), loc("PćmaŁagodnaConformance"), loc("PćmaZwyczajnaConformance")])
+    // FIXME: sourcekit returns wrong locations for the function (subclasses that don't override it, and extensions that don't implement it)
```

Comment:
Could you file a bug for this one? It seems worth understanding and fixing either in the index or in sourcekit-lsp.

Reply:
https://bugs.swift.org/browse/SR-11276